Add intro in Docs + include a "MOM6 file structure primer" section (#148)
COSIMA/regional-mom6: README.md

The idea behind this package is that it should the user sidestep some of the tricky issues with getting the model to run in the first place. This removes some of the steep learning curve for people new to working with the model. Note that the resultant model configuration might still need some tweaking (e.g., fiddling with timestep to avoid CFL-related numerical stability issues or fiddling with bathymetry to deal with very narrow fjords or channels that may exist).

> **Features**

Reviewer: ```suggestion
**Features**

```

> - Automatic grid generation at your chosen vertical and horizontal grid spacing

Reviewer: ```suggestion
- Automatic grid generation at chose vertical and horizontal grid spacing.
```

> - Finds and removes non-advective cells from your bathymetry that cause the model to crash

Reviewer: ```suggestion
- Automatic removal of non-advective cells from the bathymetry that cause the model to crash.
```

> - Handles the slicing across 'seams' in your input datasets (eg. at 0,360 or -180,180)

Reviewer: ```suggestion
- Handle slicing across 'seams' in of the forcing input datasets (e.g., when the regional configuration spans the longitude 180 of a global dataset that spans [-180, 180]).
```

> - Handles metadata encoding

Reviewer: what does this mean really? what metadata encoding?

```suggestion
- Handles metadata encoding.
```

> - Modifies pre-made configuration files to match your experiment

Reviewer: This also reads vague. "to match your experiment"? I really don't understand what the sentence is trying to say here... So we have a bunch of random config files that we modify after to comply with the configuration the user chose?

```suggestion
- Modifies pre-made configuration files to match your experiment
```

Reviewer: I think this point perhaps could be better written as:

> - Create directory structure with the configuration files as expected by MOM6

?

> - Handles interpolation and interpretation of input data. Limited pre-processing of your forcing data required!

Reviewer: This would be even stronger if no pre-processing of forcing data is required.

Author: The only pre-processing required is governed by computational limits. You could in theory pass the entire ACCESS-om2 dataset to be used as IC and BCs, but it's more stable if you first select a subset of the data and save this to disk. Although dask only reads in the metadata, I've found that if you give it a huge dataset to do complex operations on (especially when re-chunking in different directions for different boundaries is necessary) it really slows down. Perhaps there are ways around it that @angus-g might know, but I've found that generalisation with dask has its limits!

We could add another step that tries to save the subset data to disk first though? This might bring other challenges. What do you think would be best Navid?

Reviewer: Or a small rephrase:

> No pre-processing of forcing datasets is required. (In some cases, giving a subset of the forcing dataset as input helps with hitting limitations related to the machine's available memory.)

Reviewer: My point is that when I read "Limited pre-processing of your forcing data required!" to me is vague and I sort of guess that I'll have to do a bunch of manual work to make this work. But when I read what I wrote I understand what the actual limitation is.

> Limitations: Currently the package only comes with one function for generating a horizontal grid, namely one that's equally spaced in longitude and latitude. However, users can BYO a grid, or ideally open a PR with their desired grid generation function and we'll include it as an option! Further, only boundary segments parallel to longitude or latitude lines are currently supported. 

Reviewer: What’s BYO. I know but I don’t think it’s common knowledge.

Author: ```suggestion
Limitations: Currently the package only comes with one function for generating a horizontal grid, namely one that's equally spaced in longitude and latitude. However, users can provide their own grid, or ideally open a PR with their desired grid generation function and we'll include it as an option! Further, only boundary segments parallel to longitude or latitude lines are currently supported. 
```

## MOM6 Configuration and Version Requirements

> The package and demos assume a coupled SIS2-MOM6 configuration, but also works for a MOM6 ocean-only case too given appropriate changes to the `input.nml` and `MOM_input` files. For regional models, the executable must always be compiled with *symmetric* memory.

Reviewer: ```suggestion
The package and demos assume a coupled MOM6-SIS2 configuration, but also work for MOM6 ocean-only configuration after appropriate changes in the `input.nml` and `MOM_input` files.

Additionally, regional configurations require that the MOM6 executable _must_ be compiled with **symmetric memory**.
```
